library/L2_HAL/communication/test/tsop752_test.cpp

```diff
@@ -0,0 +1,154 @@
+      for (size_t i = 0; i < kFailureStatus.size(); i++)
+      {
+        INFO(
+            "Testing pulse capture initialization failure with failure status: "
+            << i);
+        const Status kExpectedStatus = kFailureStatus[i];
+        mock_pulse_capture.ClearInvocationHistory();
+        When(Method(mock_pulse_capture, Initialize))
+            .AlwaysReturn(kExpectedStatus);
+        // Upon failing to initialize the pulse capture peripheral, the failure
+        // Status should be immediately returned.
+        CHECK(ir_receiver.Initialize() == kExpectedStatus);
+        Verify(Method(mock_pulse_capture, Initialize)).Once();
+        VerifyNoOtherInvocations(mock_pulse_capture, mock_timer);
+      }
+    }
+    SECTION("Timer peripheral initialization failure")
+    {
+      When(Method(mock_pulse_capture, Initialize))
+          .AlwaysReturn(Status::kSuccess);
+
+      for (size_t i = 0; i < kFailureStatus.size(); i++)
+      {
+        INFO("Testing timer initialization failure with failure status: " << i);
+        const Status kExpectedStatus = kFailureStatus[i];
+        mock_pulse_capture.ClearInvocationHistory();
+        mock_timer.ClearInvocationHistory();
+        When(Method(mock_timer, Initialize)).AlwaysReturn(kExpectedStatus);
+
+        CHECK(ir_receiver.Initialize() == kExpectedStatus);
+        Verify(Method(mock_pulse_capture, Initialize),
+               Method(mock_pulse_capture, ConfigureCapture),
+               Method(mock_pulse_capture, EnableCaptureInterrupt),
+               Method(mock_timer, Initialize),
+               Method(mock_timer, SetMatchBehavior))
+            .Once();
+      }
+    }
+    SECTION("Both peripherals initialization success")
+    {
+      constexpr Status kExpectedStatus              = Status::kSuccess;
+      constexpr uint32_t kExpectedTimerMatchCount   = kTimeout.count();
+      constexpr uint8_t kExpectedTimerMatchRegister = 0;
+
+      mock_pulse_capture.ClearInvocationHistory();
+      mock_timer.ClearInvocationHistory();
+      When(Method(mock_pulse_capture, Initialize))
+          .AlwaysReturn(kExpectedStatus);
+      When(Method(mock_timer, Initialize)).AlwaysReturn(kExpectedStatus);
+
+      CHECK(ir_receiver.Initialize() == kExpectedStatus);
+      Verify(Method(mock_pulse_capture, Initialize)).Once();
+      Verify(Method(mock_pulse_capture, ConfigureCapture)
+                 .Using(sjsu::PulseCapture::CaptureEdgeMode::kBoth),
+             Method(mock_pulse_capture, EnableCaptureInterrupt).Using(true))
+          .Once();
+      Verify(Method(mock_timer, Initialize),
+             Method(mock_timer, SetMatchBehavior)
+                 .Using(kExpectedTimerMatchCount,
+                        sjsu::Timer::MatchAction::kInterrupt,
+                        kExpectedTimerMatchRegister))
+          .Once();
+    }
+  }
+
+  SECTION("Interrupt Callbacks")
+  {
+    constexpr std::array<uint32_t, 6> kTestTimestamps = {
```

Comment:
```suggestion
    constexpr std::array<uint32_t> kTestTimestamps = {
```
C++17 added in template deduction guides for std::array that allows you to omit the array length and allow the initializer_list<> to define it.